content/root/quick-start-guide.textile

```diff
@@ -261,15 +264,21 @@ blang[python,php].
 
 *Note*: The connection example uses basic authentication to pass the API key from the application to Ably. Basic authentication should not be used client-side in a production environment, such as in a browser, to avoid exposing the API key. "Token authentication":/core-features/authentication#token-authentication should be used instead.
 
-bc[javascript](code-editor:quick-start-guide/connect). const ably = new Ably.Realtime('{{API_KEY}}');
-ably.connection.on('connected', () => {
-  alert('Connected to Ably!');
-});
+bc[javascript]. const ably = new Ably.Realtime.Promise('{{API_KEY}}');
+try {
+  await ably.connection.once('connected');
+  console.log('Connected to Ably!');
+} catch (error) {
+  console.error(error);
```

Comment:
This catch block will never execute since the promise returned by `ably.connection.once('connected')` will never reject.

If you wanted this catch block to execute if the client fails to connect then it would have to be something like this:
```ts
await Promise.any([
  ably.connection.once('connected'),
  ably.connection.once('failed'),
]);
 ```

Reply:
I'm not sure whether using `Promise.any` will make the example too complicated for beginner JS devs. It might be worth just not catching the error and doing:
```ts
  await ably.connection.once('connected');
  console.log('Connected to Ably!');
```
If the client fails to connect the promise will never resolve and the library will still log an error message to the console so it's not like the developer wouldn't be able to see what's happening.

Reply:
I've gone for your latter suggestion. Much cleaner!